### ISSUE TITLE
I can't create table....I need help

### ISSUE THREAD

**Issue by cnzxc** [ClickHouse/ClickHouse#53780]
I can't create tables. And the error are always similar.. like this:

ClickHouse exception, code: 1002, host: 10.50.8.127, port: 8123; Code: 107. DB::Exception: There was an error on [ck1:9000]: Code: 107. DB::ErrnoException: Cannot open file /data/clickhouse/store/149/149524d6-4d04-49be-b7f7-da75aba20ef1/ai_event_mv_1386795960426496_1.sql.tmp, errno: 2, strerror: No such file or directory. (FILE_DOESNT_EXIST) (version 22.11.2.30 (official build)). (FILE_DOESNT_EXIST) (version 22.11.2.30 (official build))

The path indicated in the error is always '/data/clickhouse/149/149***'，but I always create tables with different names.  
and I can't find this path on the server disk
Anyone can help me ? please

**Comment by Tiaonmmn** [comment]
Can you share your create table sentence?

**Comment by SOLOBUCCI360** [comment]
Share your code

**Comment by cnzxc** [comment]
> Can you share your create table sentence?

This is mysql: 

-- `default`.ai_event_mv_1386155533672448 source

CREATE MATERIALIZED VIEW default.ai_event_mv_1386155533672448 TO default.ai_event_warn
(

    `event_id` String,

    `asset_id` String,

    `raw_event` String,

    `happen_time` DateTime,

    `model_id` String,

    `event_type` String,

    `cpu_usage` Float64,

    `event_name` String,

    `event_level` String,

    `risk_type` String,

    `message` String
) AS
SELECT
    generateUUIDv4() AS event_id,

    '平台' AS asset_id,

    raw_event AS raw_event,

    now() AS happen_time,

    model_id AS model_id,

    '安全事件' AS event_type,

    cpu_usage AS cpu_usage,

    'AI模型的事件' AS event_name,

    '高[7]' AS event_level,

    '网络入侵' AS risk_type,

    '发送到发' AS message
FROM
(
    SELECT
        ze.asset_id,

        ze.raw_event,

        ze.cpu_usage,

        zt.model_id AS model_id,

        zt.base_line,

        zt.e_count
    FROM default.dwd_detail_vsoc_model_event AS ze
    INNER JOIN
    (
        SELECT
            zt.model_id,

            zt.group_by,

            avg(zt.value) AS base_line,

            sum(zt.e_count) AS e_count
        FROM default.ai_statics_result AS zt
        WHERE zt.model_id = '1386155533672448'
        GROUP BY
            zt.model_id,

            zt.group_by
    ) AS zt ON zt.group_by = concat('event_level:',
 event_level)
    WHERE (ze.event_type IN ('基本事件',
 '聚合事件')) AND (event_name = 'min')
) AS seq
WHERE (seq.e_count >= 1000) AND (((seq.cpu_usage - seq.base_line) > 3.) OR ((seq.base_line - seq.cpu_usage) > 3.));

**Comment by cnzxc** [comment]
> Share your code

This is my sql: 

-- `default`.ai_event_mv_1386155533672448 source

CREATE MATERIALIZED VIEW default.ai_event_mv_1386155533672448 TO default.ai_event_warn
(

    `event_id` String,

    `asset_id` String,

    `raw_event` String,

    `happen_time` DateTime,

    `model_id` String,

    `event_type` String,

    `cpu_usage` Float64,

    `event_name` String,

    `event_level` String,

    `risk_type` String,

    `message` String
) AS
SELECT
    generateUUIDv4() AS event_id,

    '平台' AS asset_id,

    raw_event AS raw_event,

    now() AS happen_time,

    model_id AS model_id,

    '安全事件' AS event_type,

    cpu_usage AS cpu_usage,

    'AI模型的事件' AS event_name,

    '高[7]' AS event_level,

    '网络入侵' AS risk_type,

    '发送到发' AS message
FROM
(
    SELECT
        ze.asset_id,

        ze.raw_event,

        ze.cpu_usage,

        zt.model_id AS model_id,

        zt.base_line,

        zt.e_count
    FROM default.dwd_detail_vsoc_model_event AS ze
    INNER JOIN
    (
        SELECT
            zt.model_id,

            zt.group_by,

            avg(zt.value) AS base_line,

            sum(zt.e_count) AS e_count
        FROM default.ai_statics_result AS zt
        WHERE zt.model_id = '1386155533672448'
        GROUP BY
            zt.model_id,

            zt.group_by
    ) AS zt ON zt.group_by = concat('event_level:',
 event_level)
    WHERE (ze.event_type IN ('基本事件',
 '聚合事件')) AND (event_name = 'min')
) AS seq
WHERE (seq.e_count >= 1000) AND (((seq.cpu_usage - seq.base_line) > 3.) OR ((seq.base_line - seq.cpu_usage) > 3.));

**Comment by cnzxc** [comment]
> Can you share your create table sentence?

My table sql sentence: 


CREATE MATERIALIZED VIEW ai_event_mv_138685657651242414 on cluster cluster_2shards_2replicas 
to default.ai_event_warn ( 
`event_id` String, `asset_id` String, `raw_event` String, `happen_time` Datetime, `model_id` String, `event_type` String, `cpu_usage` Nullable(Float64), `event_name` Nullable(String), `event_level` Nullable(String), `risk_type` Nullable(String), `message` Nullable(String) 
) 
as select 
generateUUIDv4() as event_id, 'VSOC平台' as asset_id, raw_event as raw_event, now() as happen_time, model_id as model_id, '安全事件' as event_type, cpu_usage as cpu_usage, 'zxc' as event_name, '中[5]' as event_level, '网络入侵' as risk_type, 'zxc' as message 
from ( 
select ze.asset_id, ze.raw_event, ze.cpu_usage, zt.model_id as model_id, zt.base_line, zt.e_count 
from default.dwd_detail_vsoc_model_event ze 
global inner join ( 
select zt.model_id, zt.group_by, avg(zt.value) as base_line, sum(zt.e_count) as e_count 
from ai_event_result zt where zt.model_id = '1386851505741824' group by zt.model_id, zt.group_by 
) zt on zt.group_by = concat('event_name:',event_name,'event_level:',event_level) 
where ze.event_type in ('基本事件', '聚合事件') and ((event_name = 'zxc'  )  ) 
) seq where seq.e_count >= 1000 and (seq.cpu_usage - seq.base_line > 3.0 or seq.base_line - seq.cpu_usage > 3.0)

**Comment by cnzxc** [comment]
I tried to modify the ddl statements I used to delete and create materialized views using the dynamic data source DynamicRoutingDataSource. getDataSource(). getConnection() to obtain the connection, then using SqlRunner to execute it, and then destroying the connection and SqlRunner objects. This is how every SQL statement works. However, I have now changed it to a unified way to obtain the connection using DriverManager. getConnection, Then this connection is reused as a singleton.
